keel-rule-engine/src/main/kotlin/com/netflix/spinnaker/keel/persistence/AssetRepository.kt

```diff
@@ -19,12 +22,23 @@ interface AssetRepository {
   fun get(id: AssetId): Asset?
 
   /**
-   * Persists an asset.
+   * Fetches the ids of any assets that depend (directly) on [id].
+   * Get a partial asset
    */
-  fun store(asset: Asset)
+  fun getPartial(id: AssetId): PartialAsset?
```

Comment:
Does it make sense to have a distinct id type?

Reply:
You tell me. I don't think it matters, but I'm happy to add it in.

Reply:
I guess let's keep it the same for now & consider it separately from this PR.